### PR TITLE
Skip batch requests for unconnected modules

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #2358 

## Relevant technical choices

* Checks module setup before no data since skipped requests will always have no data and `setupComplete` is available  on page load
* Added basic tests for `withData`

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/higherorder/withdata.js [hunk 1]
```diff
@@ -219,34 +221,32 @@ const withData = (
...
-						 * Request data for the context.
-						 */
-						addFilter( `googlesitekit.module${ acontext }DataRequest`,
-							`googlesitekit.data${ acontext }`, ( moduleData ) => {
-								data.callback = ( returnedData ) => {
-									handleReturnedData( returnedData, data );
-								};
-								moduleData.push( data );
-								return moduleData;
-							} );
```
Comment (aaemnnosttv) [inline assets/js/components/higherorder/withdata.js:235]: This was previously duplicated for both the array and non-array cases. It's been consolidated into a single block using `castArray`.

**Review by felixarntz — CHANGES_REQUESTED** [review]
Technically this works fine, however it results in some odd problem that now the "All Traffic" widget (legacy) displays the "Complete activation" CTA three times instead of just once for the entire widget. This is likely due to some detail in that component and how it makes its requests, but we should fix it to avoid a regression due to this fix.

Screenshot 2020-11-18 at 11 34 01

**Comment by aaemnnosttv** [comment]
@felixarntz it had to do specifically with how the `LegacyAnalyticsAllTraffic` component rendered the error/incomplete-setup component separately and had separate logic for hiding the inner components in that case (`.googlesitekit-nodata`) which was slightly different than what was used to select the CTA component which made it possible for them to both show. I've updated that component to conditionally render the inner components now rather than simply hiding with CSS. Now that requests for modules with incomplete setup are no longer made, that means that components rendered `withData` that only request the single module and use the `handleDataError` or `handleDataSuccess` callbacks, these will never be called because they rely on a data response. I don't think this should really cause a problem since it only affects the dashboard which is easy to check. The module dashboard isn't accessible because it will just load the module setup, so I don't foresee this "sneaking up on us" later somewhere else.

image

As an aside, I also made a small tweak to `withData` so that it no longer hides the wrapped component names and shows a [proper display name](https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging) for an HOC, which is really refreshing.

image

**Review by felixarntz — APPROVED** [review]
Great! Looks like this also fixed another (even more severe) issue on the single URL details view: In there, the "All Traffic" widget used to stay in loading state forever, due to a JS error (this is even broken in the last release, and probably has been for a while 😱 ):
Screenshot 2020-11-18 at 15 40 30

This PR fixes it so the CTA shows there instead. (It does currently show twice, which _may_ be worth improving, but I'd rather ignore that for now, since what we have here is a huge improvement. Let's just get it into the release as is.)

**Pull request merged**